### PR TITLE
573 handle id_resp and generate id_result for frontend

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Handle the id response from Sage properly
- Build the ID_result for the frontend

---

**REST API Updates **

```
[GET] /api/v1/sightings/{sighting_uid}/id_result
{
    "annotation_data": {
        "5c9bc1b2-5e5f-4029-88e4-43e6d94c5bf3": {
            "asset_dimensions": {
                "height": 664,
                "width": 1000
            },
            "bounds": {
                "rect": [
                    45,
                    5,
                    78,
                    3
                ],
                "theta": 4.8
            },
            "encounter_location": null,
            "image_url": "http://localhost/api/v1/assets/src/764cc968-d3b7-44d5-a802-db9b79095608",
            "individual_guid": null,
            "viewpoint": "test"
        },
        "7e0460b0-dac7-43cb-9e0e-c19988b31301": {
            "asset_dimensions": {
                "height": 664,
                "width": 1000
            },
            "bounds": {
                "rect": [
                    45,
                    5,
                    78,
                    3
                ],
                "theta": 4.8
            },
            "encounter_location": null,
            "image_url": "http://localhost/api/v1/assets/src/54bb541e-3217-4938-9c75-2278870bc7ff",
            "individual_guid": null,
            "viewpoint": "test"
        }
    },
    "individual_data": {},
    "query_annotations": [
        {
            "algorithms": {
                "hotspotter_nosv": {
                    "scores_by_annotation": [
                        {
                            "guid": "5c9bc1b2-5e5f-4029-88e4-43e6d94c5bf3",
                            "id_finish_time": "2022-01-12 18:46:29",
                            "score": 0.49249419758140284
                        }
                    ],
                    "scores_by_individual": [
                        {
                            "guid": "5c9bc1b2-5e5f-4029-88e4-43e6d94c5bf3",
                            "id_finish_time": "2022-01-12 18:46:29",
                            "score": 4.755731035123042
                        }
                    ]
                }
            },
            "guid": "7e0460b0-dac7-43cb-9e0e-c19988b31301",
            "individual_guid": null,
            "status": "complete"
        }
    ]
}

```

### CODE REVIEW

File: app/extensions/acm/__init__.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def from_acm_uuid(uuid_str):
     import uuid
 
     assert '__UUID__' in uuid_str.keys()
-    return uuid.uuid4(uuid_str['__UUID__'])
+    return uuid.UUID(uuid_str['__UUID__'])
```
Comment (hwindsor) [inline app/extensions/acm/__init__.py:29]: This was a bug

File: app/modules/annotations/models.py [hunk 2]
```diff
@@ -147,6 +147,10 @@ def set_bounds(self, bounds):
         self.validate_bounds(bounds)
         self.bounds = bounds
 
+    def get_image_url(self):
+        # TODO this needs a new Jira
```
Comment (hwindsor) [inline app/modules/annotations/models.py:151]: Or does it, ongoing discussion at the moment, this is the primary reason this is still draft.

File: app/modules/encounters/models.py [hunk 3]
```diff
@@ -114,6 +114,16 @@ def get_owner(self):
     def get_sighting(self):
         return self.sighting
 
+    def get_location(self):
+        edm_data = current_app.edm.get_dict('encounter.data_complete', self.guid)
+        location_id = None
+        if isinstance(edm_data, dict) and edm_data.get('success', False):
+            edm_json = edm_data['result']
+
+            if 'locationId' in edm_json.keys():
```
Comment (hwindsor) [inline app/modules/encounters/models.py:123]: No idea if this is the correct way to get the location. I suspect it isn't.

File: app/modules/sightings/resources.py [hunk 4]
```diff
@@ -677,9 +677,9 @@ def post(self, sighting):
     description='Sighting not found.',
 )
 @api.resolve_object_by_model(Sighting, 'sighting')
-class SightingIdentified(Resource):
+class SightingSageIdentified(Resource):
     """
-    Detection of Asset Group Sighting complete
+    Indentification of Sighting complete
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:682]: tidying up some cruft as passing

File: tests/modules/sightings/resources/utils.py [hunk 5]
```diff
@@ -351,50 +350,42 @@ def delete_sighting(
 
 
 # Create a default valid Sage detection response (to allow for the test to corrupt it accordingly)
-def build_sage_identification_response(job_uuid, annot_uuid, algorithm):
+def build_sage_identification_response(
+    job_uuid, annot_uuid, algorithm, target_annot_uuid
+):
 
     # Generate the response back from Sage.
-    # 15th July, Do not assume this is correct. This is hacky first try and probably needs
```
Comment (hwindsor) [inline tests/modules/sightings/resources/utils.py:357]: It was wrong, very wrong

**Review by colinwkingen — APPROVED** [review]
LGTM

**Comment by colinwkingen** [comment]
@brmscheiner Please take a peek at the summary here before I merge the PR in.

**Comment by brmscheiner** [comment]
@colinwkingen I'm sure I'll realize some things when I start working with it, but at first glance this looks great. Does this PR also cause ID to happen?

**Comment by colinwkingen** [comment]
@brmscheiner Nope, but it's a step on the way. Thanks!

**Pull request merged**